### PR TITLE
Task 7 - Add/Edit/Delete movie with Formik

### PR DESCRIPTION
**Pull request by apyd**
Done:

- [x] - Installed formik
- [x] - Implemented “Add movie” form with validation
- [x] - Implemented “Edit movie” form with validation. Used hooks from formik
- [x] - Implemented “Delete movie” functionality

Not done: N/A

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { Hero } from "../Hero/Hero";
+import { Results } from "../Results/Results";
+import { Footer } from "../Footer/Footer";
+
+import styles from "./App.scss";
+import { Provider } from "react-redux";
+import { store } from "../../store/store";
+
+export const App = () => {
+  const cx = classNames.bind(styles);
```
Comment (Vladlan) [inline src/components/App/App.tsx]: Move const cx = classNames.bind(styles); out of the component's scope

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import classNames from "classnames/bind";
+import { Logo } from "../Logo/Logo";
+import { LogoSize } from "../Logo/Logo.const";
+import styles from "./Footer.scss";
+
+export const Footer = () => {
+  const cx = classNames.bind(styles);
```
Comment (Vladlan) [inline src/components/Footer/Footer.tsx]: the same here

File: src/store/api/apiTypes.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export interface Movie {
```
Comment (Vladlan) [inline src/store/api/apiTypes.ts]: movie.interface.ts

**Review by Vladlan — CHANGES_REQUESTED** [review]
The task is not ready. Please review requirements and add changes. Also address comments from Task 6. Only after approve of task 7 and merge start Task 8 to make review clean (I want to see changes only related to the task and not entire app).

File: src/components/Modal/MovieForm/configuration.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import * as Yup from 'yup';
+
+export const FormSchema = Yup.object().shape({
+  title: Yup.string()
+    .min(2, "Title can't be shorter than 2 characters!")
+    .max(50, "Exceeded limit of characters - 50!")
+    .required("This field is required"),
+  poster_path: Yup.string()
+    .matches(
+      /((https?):\/\/)?(www.)?[a-z0-9]+(\.[a-z]{2,}){1,3}(#?\/?[a-zA-Z0-9#]+)*\/?(\?[a-zA-Z0-9-_]+=[a-zA-Z0-9-%]+&?)?$/,
```
Comment (Vladlan) [inline src/components/Modal/MovieForm/configuration.ts]: Remove www from validation

**Comment by apyd** [comment]
Comment with pending changes to be done from previous PR: https://github.com/apyd/Movies-app/pull/4#issuecomment-1173709872

**Review by Vladlan — CHANGES_REQUESTED** [review]
Please fix React red warning:

Screenshot 2022-07-28 at 18 28 35

**Review by Vladlan — APPROVED** [review]

**Pull request merged**